### ISSUE TITLE
Clarification needed on learning path repository to use, "Open-Source-Orchard-Core-Extensions" or "Orchard-Training-Demo-Module"? (OSOE-783)

### ISSUE THREAD

**Issue by MarGraz** [Lombiq/Orchard-Training-Demo-Module#128]
Hi everyone,

I'm currently following the learning path for Orchard Core and have encountered some confusion regarding the repositories involved.

Initially, I cloned the "[Open-Source-Orchard-Core-Extensions](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions)" repository as indicated in the "How to start" step [here](https://github.com/Lombiq/Orchard-Training-Demo-Module?tab=readme-ov-file#how-to-start) (I did a [pull request](https://github.com/Lombiq/Orchard-Training-Demo-Module/pull/125) two days ago about this). The "Open-Source-Orchard-Core-Extensions" repository is the one contains the `Lombiq.OSOCE.Web` solution and project. Keep in mind for later that `Lombiq.OSOCE.Web` is not present in the "[Orchard-Training-Demo-Module](https://github.com/Lombiq/Orchard-Training-Demo-Module)" repository.

Upon reaching the "[Learning Orchard Core with this module](https://github.com/Lombiq/Orchard-Training-Demo-Module?tab=readme-ov-file#learning-orchard-core-with-this-module)" section, I was invited to follow the "[StartLearningHere.md](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/StartLearningHere.md)" file. In the [Let's get started](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/StartLearningHere.md#lets-get-started) section, it mentions to _"look into the Dependencies **of this project** you will find either the NuGet reference for the OrchardCore.Module.Targets package"_. However, I couldn't find any reference to this package in the `Lombiq.OSOCE.Web` project from the "Open-Source-Orchard-Core-Extensions" repository previously cloned. Instead, this reference appears to be in the "Orchard-Training-Demo-Module".

Could you please clarify which repository and branch should be used to properly follow the learning path? To follow "[StartLearningHere.md](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/StartLearningHere.md)" I need to clone another repo? The discrepancy between the `Lombiq.OSOCE.Web` project in the "Open-Source-Orchard-Core-Extensions" and the "Orchard-Training-Demo-Module" is quite confusing.

Thank you so much.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-783)

**Comment by Piedone** [comment]
In the Training Demo, every "this project" refers to, well, this project :). `OrchardCore.Module.Targets` is a reference of the Training Demo module. I don't really understand how you confused that with the `Lombiq.OSOCE.Web` project; that's only interesting to set as a startup project in the OSOCE solution, if necessary.

**Comment by MarGraz** [comment]
Thank you, @Piedone, for your reply.

I got confused because in the "[How to start](https://github.com/Lombiq/Orchard-Training-Demo-Module?tab=readme-ov-file#how-to-start)," at step 2, the readme mentions `Lombiq.OSOCE.Web`, which is only present in the OSOCE solution. However, when I visit [StartLearningHere.md](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/StartLearningHere.md), instead it seems to refer to the [Orchard-Training-Demo-Module](https://github.com/Lombiq/Orchard-Training-Demo-Module).

So, to follow [StartLearningHere.md](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/StartLearningHere.md), do I need to clone and run the [Orchard-Training-Demo-Module](https://github.com/Lombiq/Orchard-Training-Demo-Module), right?

If this is correct, I will add a note near the "StartLearningHere.md" link in "[Learning Orchard Core with this module](https://github.com/Lombiq/Orchard-Training-Demo-Module?tab=readme-ov-file#learning-orchard-core-with-this-module)." This way, users will be aware that a step before, we ask them to clone the latest dev version of "Open-Source-Orchard-Core-Extensions".

Thank you

**Comment by Piedone** [comment]
The idea here is that you clone the OSOCE repo, since it includes the Training Demo as a Git submodule and everything else you need for it. So, no need to separately clone the Training Demo; instead, you get a full solution with an Orchard Core app (that's necessary, since you can't work with a module in itself). Perhaps we could emphasize all this in that docs section?

Thank you!

If you have a similar attention to detail in your daily work too, then we'd be happy to hire you once you learned Orchard Core, just email me under zoltan.lehoczky@lombiq.com :).

**Comment by MarGraz** [comment]
Thank you @Piedone for your reply.

I can add a note in the [Learning Orchard Core with this module](https://github.com/Lombiq/Orchard-Training-Demo-Module?tab=readme-ov-file#learning-orchard-core-with-this-module) section, where I will explain that the guide will analyze each module project and mention that these modules can be found in the `src` > `Modules` folder of the `Lombiq.OSOCE` solution.

Thank you for your compliments and the offer 😀. Yes, usually I have this attention to detail 😊. I've actually started a company with a friend here in the Netherlands, [CodingOps](https://codingops.dev). However, I will definitely send you an email in the future for potential collaboration and to share what we have accomplished using Orchard.

Thank you so much
